src/test/ui/issues/issue-20413.stderr

```diff
@@ -87,7 +88,86 @@ note: required by `Foo`
 LL | trait Foo {
    | ^^^^^^^^^
 
-error: aborting due to 2 previous errors
+error[E0275]: overflow evaluating the requirement `NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<T>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>: Foo`
```

Comment:
I wonder if this constitutes an error reporting regression. It seems like a lot of spam. cc @estebank 
(do we have a GH team for diagnostics questions?)

Reply:
There's already a similar error in this file. This just happens in two locations now since we do them in parallel and don't stop at the first one.

Reply:
> I wonder if this constitutes an error reporting regression

I would say it does. In the past for similar situations where there's no way of reworking the current diagnostic plumbing, a way we've avoided duplicated errors is by keeping around a `RefCell<HashSet>` in the `Session` keeping track of all the `Span`/`NodeId`/`HirId` that have already been emitted and avoid emitting them a second time.

> It seems like a lot of spam.

It does, and I'd go as far as saying that the current error is already quite bad, even without it triggering multiple times. There have been some efforts in the past to identify cycles and make the errors less verbose, but this case of `NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<NoData<T>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>` isn't yet handled, and it should.

I understand @Zoxc if you don't want to deal with these things in this PR, but would you mind filing follow up issues for the two points raised above if you don't, just so we don't lose track of them?

> (do we have a GH team for diagnostics questions?)

We don't and we probably should, as there're quite a few of us now. I'll bring it up the next compiler meeting.

Reply:
Opened https://github.com/rust-lang/rust/issues/58964 and https://github.com/rust-lang/rust/issues/58965